src/Core/System/Snippet/Struct/TranslationFile.php

```diff
@@ -0,0 +1,129 @@
+<?php declare(strict_types=1);
+
+namespace Shopware\Core\System\Snippet\Struct;
+
+use Shopware\Core\Framework\Log\Package;
+use Shopware\Core\Framework\Struct\Struct;
+
+/**
+ * @internal
+ */
+#[Package('discovery')]
+class TranslationFile extends Struct
+{
+    public function __construct(
+        protected string $filename,
```

Comment:
why protected? you could change it to `public readonly` and you could delete all getter and setter here. The setter are also not used.

Reply:
DTOs extending from `Struct` are mostly used to be json_encoded and given to an API. as this is not the case here, I would go with plain classes extending nothing and with the public readonly properties